pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -215,25 +240,107 @@ func (r *CloudProviderAccountReconciler) startPollingThread(namespacedName *type
+func (r *CloudProviderAccountReconciler) getCpaBySecret(s types.NamespacedName) (error, []crdv1alpha1.CloudProviderAccount) {
+	cpaList := &crdv1alpha1.CloudProviderAccountList{}
+	if err := r.Client.List(context.TODO(), cpaList, &client.ListOptions{}); err != nil {
+		return fmt.Errorf("failed to get CloudProviderAccount list, err %v", err), nil
+	}
+	var cpaItems []crdv1alpha1.CloudProviderAccount
+	for _, cpa := range cpaList.Items {
+		if cpa.Spec.AWSConfig != nil {
+			if cpa.Spec.AWSConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AWSConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+
+		if cpa.Spec.AzureConfig != nil {
+			if cpa.Spec.AzureConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AzureConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+	}
+	return nil, cpaItems
+}
+
+// watchSecret watch the Secret objects.
+func (r *CloudProviderAccountReconciler) watchSecret() {
+	for {
+		event, ok := <-r.watcher.ResultChan()
+		if !ok {
+			r.resetSecretWatcher()
+		} else {
+			switch event.Type {
+			case watch.Modified:
+				r.mutex.Lock()
+				secret := event.Object.(*v1.Secret)
+				namespacedName := types.NamespacedName{Namespace: secret.Namespace, Name: secret.Name}
+				r.Log.WithName("Secret").Info("Received request", "Secret", namespacedName, "operation", "update")
+				err, cpaItems := r.getCpaBySecret(namespacedName)
+				if err != nil {
+					r.Log.WithName("Secret").Error(err, "error getting CPA by Secret", "account", namespacedName)
+				}
+				for _, cpa := range cpaItems {
+					// If the credentials are updated to new cloud account then vpc cache is not updated. As a
+					// result vpc cache will contain vpcs from old and new accounts.
+					accountNamespacedName := types.NamespacedName{Namespace: cpa.Namespace, Name: cpa.Name}
+					err := r.processCreateOrUpdate(&accountNamespacedName, &cpa)
+					if err != nil {
+						r.Log.WithName("Secret").Error(err, "error updating account", "account", accountNamespacedName)
+					} else {
+						r.Log.WithName("Secret").Info("Done processing secret", "account", accountNamespacedName)
+					}
+				}
+				r.mutex.Unlock()
+			case watch.Error:
+				r.resetSecretWatcher()
+			}
+		}
+	}
+}
+
+// resetSecretWatcher create a watcher for Secret
+func (r *CloudProviderAccountReconciler) resetSecretWatcher() {
+	var err error
+	for {
+		if r.watcher, err = r.clientset.CoreV1().Secrets(r.getPodNamespace()).Watch(context.Background(), metav1.ListOptions{}); err != nil {
+			r.Log.WithName("Secret").Error(err, "error creating Secret watcher")
+			time.Sleep(time.Second * 5)
+			continue
+		}
+		break
+	}
+}
+
+// setupSecretWatcher set up a watcher for Secret objects.
+func (r *CloudProviderAccountReconciler) setupSecretWatcher() {
+	r.resetSecretWatcher()
+	r.watchSecret()
+}
+
+// getPodNamespace gets the namespace of the pod.
+func (r *CloudProviderAccountReconciler) getPodNamespace() string {
```

Comment:
I dont think this is required here.. You should check for empty ns in GetPodNamespace. not here.. Also verify if nephe is launched in default Namespace, whats the value you get.

Reply:
This is required because the GetPodNamespace is generic so we should not handle it there. Because in this case we exit the pod. 
The same namespace value which is configured in the pod metadata.